### PR TITLE
Make widget configs easy to override

### PR DESCRIPTION
**Pull request by clpetersonucf**
Moves some widget configurations out of `getLegacyWidgetBuildConfig`, allowing them to be more easily referenced/overridden: 
- The default copyList is returned via `getDefaultCopyList()`, which can be modified, appended, and used to override the copyList sent to the copyPlugin by default.
- JS rules have been encapsulated into their own object allowing them to be individually referenced, modified, or removed.
- Plugins have also been moved into an object allowing individual steps to be changed.

### CODE REVIEW

File: webpack-widget.js [hunk 1]
```diff
@@ -91,6 +85,141 @@ const getLegacyWidgetBuildConfig = (config = {}) => {
 			toType: 'dir'
 		}
 	]
+}	
+
+const getDefaultJSRules = () => {
```
Comment (iturgeon) [inline webpack-widget.js]: this guy's name might be a little misleading, it contains css/sass rules too

File: webpack-widget.js [hunk 2]
```diff
@@ -46,24 +46,18 @@ const configFromPackage = () => {
...
-// you can update the return from this method to modify or alter
-// the base configuration
-const getLegacyWidgetBuildConfig = (config = {}) => {
-	// load and combine the config
+// Provides a default config option
+const getDefaultCfg = (extras = {}) => {
 	let materiaConfig = configFromPackage()
-	cfg = Object.assign({}, defaultCfg, {cleanName:materiaConfig.cleanName}, config)
-	// set up source and destination paths
+	const configs = Object.assign({}, defaultCfg, {cleanName:materiaConfig.cleanName}, extras)
```
Comment (iturgeon) [inline webpack-widget.js]: just return here

File: webpack-widget.js [hunk 3]
```diff
@@ -102,6 +231,19 @@ const getLegacyWidgetBuildConfig = (config = {}) => {
 		})
 	}
 
+	let rules = getDefaultJSRules()
+
+	let plugins = {
```
Comment (iturgeon) [inline webpack-widget.js]: This seems half implemented.  There's no point to doing this and then using them in the rules section later unless we want to export them so they can be referenced. 

Either make sure they are exported for use or just define them inline in the rules section below

Comment (FrenjaminBanklin) [reply webpack-widget.js]: Is this to preserve the current expected default functionality?

File: webpack-widget.js (hunk 1 above)
Comment (FrenjaminBanklin) [reply webpack-widget.js]: Agree. "defaultRules" might be more fitting.

**Pull request merged**